### ISSUE TITLE
FastAPI unknown import symbol

### ISSUE THREAD

**Issue by Andarius** [microsoft/pyright#2783]
**Describe the bug**
When running `pyright test.py`, the following errors are raised:

```
test.py
  test.py:1:21 - error: "Response" is unknown import symbol (reportGeneralTypeIssues)
  test.py:2:32 - error: "TestClient" is unknown import symbol (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 infos 
```

where **test.py** is:

```python
from fastapi import Response
from fastapi.testclient import TestClient
```

**To Reproduce**

Create the test environment: 

```bash
conda create -n test python=3.9 fastapi pyright -c conda-forge
conda activate test
```

then run 

```bash
pyright test.py
```

**Expected behavior**

To not show any errors

**VS Code extension or command-line**

`pyright test.py`

**Comment by erictraut** [comment]
Which version of fastapi are you using? I'm not able to repro this issue with the latest published version (0.70.1).

**Comment by Andarius** [comment]
Sorry I did not mention it, I'm using the latest conda-forge version available: `0.70.1`

**Comment by erictraut** [comment]
The problem is that `fastapi` is importing types (and then re-exporting them) from a library called `starlette`. While `fastapi` is fully typed, the version of `starlette` that is installed in your conda environment is not. The version installed in your conda environment is "0.16.0", but the latest version (which does include inlined types) is version "0.17.1".

**Comment by Andarius** [comment]
`fastapi` ships with the `0.16.0` (https://github.com/tiangolo/fastapi/blob/3efb4f7edff99fdc12802a85ae9d140ec4772497/pyproject.toml#L36) version of starlette as hard dependency and is not compatible with the 0.17.1 version.
What is the best way to avoid this error message ?

**Comment by erictraut** [comment]
You can enable `useLibraryCodeForTypes`, which tells pyright to do its best to infer type information from `starlette` and other typed libraries, but your results may vary.

Add this to your pyrightconfig.json file:
```
{
  "useLibraryCodeForTypes": true
}
```
